File: Deliverable2/APPS/DESKTOP/TAMAS/src/ca/mcgill/ecse321/group10/controller/CourseController.java
package ca.mcgill.ecse321.group10.controller;

import ca.mcgill.ecse321.group10.TAMAS.model.Course;
import ca.mcgill.ecse321.group10.TAMAS.model.CourseManager;
import ca.mcgill.ecse321.group10.persistence.PersistenceXStream;

public class CourseController {
	
	public static String COURSE_FILE_NAME = "output/courses.xml";

	private CourseManager cm;
	
	public CourseController(CourseManager cm) {
		this.cm = cm;
	}
	
	public void createCourse(String aClassName, int aCdn, float aGraderTimeBudget, float aTaTimeBudget) throws InputException {
		String error = "";
		if(aClassName == null || aClassName.trim().length() == 0) {
			error += "Course name cannot be empty! ";
		}
		if(error.length() > 0) throw new InputException(error);
		Course c = new Course(aClassName,aCdn,aGraderTimeBudget,aTaTimeBudget);
		cm.addCourse(c);
		PersistenceXStream.setFilename(COURSE_FILE_NAME);
		PersistenceXStream.saveToXMLwithXStream(cm);
	}
}

File: Deliverable2/APPS/WEB/TAMAS/ca.mcgill.ecse321.group10.TAMAS/Controller/CourseController.php
<?php
require_once __DIR__.'\..\Controller\InputValidator.php';
require_once __DIR__.'\..\Persistence\PersistenceTAMAS.php';
require_once __DIR__.'\..\Model\CourseManager.php';
require_once __DIR__.'\..\Model\Course.php';

class CourseController{
	private $pt = new PersistenceTAMAS();
	private $cm = $pt->loadCourseManagerFromStore();
				
	public function __construct(){
	}
	
	public function createCourse($course_name, $CDN, 
								$graderTimeBudget, $TATimeBudget) {
		// Validate input
		$error = "";
		$name = InputValidator::validate_input($course_name);
		if($name==null || strlen($name) == 0){
			$error .= ("Course name cannot be empty!<br> ");
		} 
		if(!is_numeric($CDN)) {
			$error .= ("CDN must be a non null Integer!<br> ");
		} 
		if((!is_numeric($graderTimeBudget)) || (!is_numeric($TATimeBudget))) {
			$error .= ("Time budget must be a non null Integer!<br> ");
		}
		
		if(strlen($error) > 0) {
			throw new Exception($error);
		} else {
			// Add the new course
			$course = new Course($name, $CDN, $graderTimeBudget, $TATimeBudget);
			$this->cm->addCourse($course);
				
			//4. Write all the data
			$this->pt->writeCourseDataToStore($this->cm);
		}
	}

}
?>